src/system/SystemClock.h

```diff
@@ -187,96 +77,57 @@ class Clock
-     * @retval #CHIP_ERROR_REAL_TIME_NOT_SYNCED
-     *                                      If the platform is capable of tracking real time, but is
-     *                                      is currently unsynchronized.
-     * @retval #CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE
-     *                                      If the platform is incapable of tracking real time.
-     */
-    static inline CHIP_ERROR GetUnixTimeMicroseconds(UnixTimeMicroseconds & curTime)
-    {
-        // Current implementation is a simple pass-through to the platform.
-        return Platform::Clock::GetUnixTimeMicroseconds(curTime);
-    }
+// Currently we have a single implementation class, ClockImpl, whose members are implemented in build-specific files.
+class ClockImpl : public ClockBase
+{
+public:
+    ~ClockImpl() = default;
+    MonotonicMicroseconds GetMonotonicMicroseconds() override;
+    MonotonicMilliseconds GetMonotonicMilliseconds() override;
+};
 
-    /**
-     * Sets the platform's notion of current real (civil) time.
-     *
-     * Applications can call this function to set the local platform's notion of current real time.  The
-     * new current time is expressed as a Unix time value scaled to microseconds.
-     *
-     * Once set, underlying platform clock is guaranteed to track real time with a granularity of at least
-     * whole seconds.
-     *
-     * Some platforms may restrict which applications or processes can set real time.  If the caller is
-     * not permitted to change real time, the SetRealTime() function will return the error
-     * CHIP_ERROR_ACCESS_DENIED.
-     *
-     * On platforms that are incapable of tracking real time, or do not offer the ability to set real time,
-     * the SetRealTime() function may be absent, resulting a link error for any application that
-     * references it.  Alternatively, such platforms may supply an implementation of SetRealTime()
-     * that always returns the error CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE.
-     *
-     * This function is guaranteed to be thread-safe on any platform that employs threading.
-     *
-     * @param[in] newCurTime                The new current time, expressed as Unix time scaled to microseconds.
-     *
-     * @retval #CHIP_NO_ERROR               If the method succeeded.
-     * @retval #CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE
-     *                                      If the platform is incapable of tracking real time.
-     * @retval #CHIP_ERROR_ACCESS_DENIED
-     *                                      If the calling application does not have the privilege to set the
-     *                                      current time.
-     */
-    static inline CHIP_ERROR SetUnixTimeMicroseconds(UnixTimeMicroseconds newCurTime)
-    {
-        // Current implementation is a simple pass-through to the platform.
-        return Platform::Clock::SetUnixTimeMicroseconds(newCurTime);
-    }
+extern ClockImpl staticClock;
```

Comment:
nit: should these have a `g` prefix since we seem to use a somewhat  hugarian notation in other places (m for members, g for globals)? Although admittedly `gGlobalClock` sounds a  bit odd ... maybe `gClockBase` and `gClockImpl`.

Reply:
In this case I was avoiding the normal convention to suggest that the variable should never be accessed directly, but really I should have just written that explicitly. The System::Layer PR has the same pattern. I'll make followups if these land as-is.

Reply:
As a followup, consider using a namespace Internal - I have seen this pattern before for things that are not to be used directly unless one knows what they are doing